cylc/flow/scheduler.py

```diff
@@ -1230,11 +1231,150 @@ def remove_tasks(
             tasks_str = ', '.join(
                 sorted(tokens.relative_id for tokens in not_removed)
             )
-            LOG.warning(f"Task(s) not removable: {tasks_str} {fnums_str}")
+            # This often does not indicate an error - e.g. for group trigger.
+            LOG.debug(f"Task(s) not removable: {tasks_str} {fnums_str}")
 
         if removed and self.pool.compute_runahead():
             self.pool.release_runahead_tasks()
 
+    def force_trigger_tasks(
+        self,
+        items: Iterable[str],
+        flow: List[str],
+        flow_wait: bool = False,
+        flow_descr: Optional[str] = None,
+        on_resume: bool = False
+    ):
```

Comment:
Co-locate this with the other commands in `cylc.flow.commands`.

Reply:
> It is! 

???

This code has been yanked out into the Scheduler class for some reason, it should be co-located with the other commands in `cylc.flow.commands`.

Reply:
Oh, I see what you mean.  

It was yanked out of the task_pool module, not commands - just as @MetRonnie did to the remove command.

Both trigger and remove have a small wrapper in commands.  I think I was just following Ronnies example, since the two commands have relevant similarities.

Reply:
I didn't remove anything from `commands.py` in #6472, just moved it from `task_pool.py` to `scheduler.py` as it needed to call the scheduler's `kill_tasks` method. It looks like you've done a similar thing and need to call the scheduler's `remove_matched_tasks` method

Reply:
Correct!

Reply:
Er, that doesn't actually make sense in either case.

The commands are provided with `schd` as the first argument. This is exactly the same as the `self` argument! The code does not need to be moved into the Scheduler to call scheduler methods.

Reply:
What he said. 

Like for remove, trigger was moved out of task_pool to scheduler. It was never in commands.

In both cases, there was just a small wrapper in commands that called the task pool (now scheduler) methods.

Reply:
Done.